pkg/apis/pipeline/v1beta1/task_validation.go

```diff
@@ -242,16 +242,6 @@ func validateStep(ctx context.Context, s Step, names sets.String) (errs *apis.Fi
 		}
 	}
 
-	if s.OnError != "" {
-		if s.OnError != "continue" && s.OnError != "stopAndFail" {
-			errs = errs.Also(&apis.FieldError{
-				Message: fmt.Sprintf("invalid value: %v", s.OnError),
-				Paths:   []string{"onError"},
-				Details: "Task step onError must be either continue or stopAndFail",
-			})
-		}
-	}
-
```

Comment:
NIT: I think we could still do an early validation is no variables are used in `onError` - and keep the test below?

Reply:
sure sounds good, I will add them back